paper-server/patches/sources/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -984,7 +984,11 @@
 +            if (invulnerabilityRelatedLastDamage == 0) return 0D; // no last damage, no reduction
 +            // last damage existed, this means the reduction *technically* is (new damage - last damage).
 +            // If the event damage was changed to something less than invul damage, hard lock it at 0.
-+            if (d < invulnerabilityRelatedLastDamage) return 0D;
++            //
++            // Cast the passed in double down to a float as double -> float -> double is lossly.
++            // If last damage is a (float) 3.2D (given spigot events use doubles), we cannot compare
```

Comment:
```suggestion
+            // Cast the passed in double down to a float as double -> float -> double is lossy.
+            // If last damage is a (float) 3.2D (since the events use doubles), we cannot compare
```